internal/handler/handler_test.go

```
@@ -0,0 +1,746 @@
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/shopspring/decimal"
+)
+
+var (
+	data        *repository.Database   //nolint: gochecknoglobals
+	repos       *repository.Repository //nolint: gochecknoglobals
+	serv        *service.Serv          //nolint: gochecknoglobals
+	handler     *Hand                  //nolint: gochecknoglobals
+	positionIDs []string               //nolint: gochecknoglobals
+	employeeIDs []string               //nolint: gochecknoglobals
+)
+
+func initTest() {
+	data = repository.NewDataBase()
+	repos = repository.NewRepo(data)
+	serv = service.NewServ(repos)
+	handler = NewHandler(serv)
+	positionIDs = make([]string, 0)
+	employeeIDs = make([]string, 0)
+}
+
+func createPosID() uuid.UUID {
+	id := uuid.New()
+	positionIDs = append(positionIDs, id.String())
+	return id
+}
+
+func createEmpID() uuid.UUID {
+	id := uuid.New()
+	employeeIDs = append(employeeIDs, id.String())
+	return id
+}
+
+func createTestContext(r *http.Request) *http.Request {
+	ctx := r.Context()
+	id := uuid.New()
+	//revive:disable
+	ctx = context.WithValue(ctx, "correlation_id", id.String()) //nolint:staticcheck
+	//revive:enable
+	r = r.WithContext(ctx)
+	return r
+}
+
+func TestHand_CreateEmployee(t *testing.T) {
+	initTest()
+	firstPosition := internal.Position{ID: createPosID(), Salary: decimal.New(500, 0), Name: "worker"}
+	repos.AddPosition(&firstPosition)
+	id, err := uuid.Parse(positionIDs[0])
+	if err != nil {
+		t.Fatalf("Error: %v ", err)
+	}
+	firstEmployee := internal.Employee{ID: createEmpID(), FirstName: "Bob", LasName: "Vik", PositionID: id}
+	secondEmployee := internal.Employee{ID: createEmpID(), FirstName: "", LasName: "Vik", PositionID: id}
+	thirdEmployee := internal.Employee{ID: createEmpID(), FirstName: "Fox", LasName: "Fok", PositionID: uuid.New()}
+	jsonFirstEmployee, _ := json.Marshal(firstEmployee)
+	jsonSecondEmployee, _ := json.Marshal(secondEmployee)
+	jsonThirdEmployee, _ := json.Marshal(thirdEmployee)
```

Ignoring errors is a bad practice

---

+